File: clojure/cljfind/src/cljfind/color.clj
;;; ############################################################################
;;;
;;; color.clj
;;;
;;; Defines generic colors
;;;
;;; ############################################################################

(ns cljfind.color
  #^{:author "Cary Clark",
     :doc "Defines generic colors"}
  (:use [clojure.string :as str :only (lower-case)]
        [cljfind.consolecolor]))

;; color keyword names
(def ^:const COLOR_NAMES [:black :red :green :yellow :blue :magenta :cyan :white])

(defn get-console-color-for-color [c]
  (cond
    (= :black c) CONSOLE_COLOR_BLACK
    (= :red c) CONSOLE_COLOR_RED
    (= :green c) CONSOLE_COLOR_GREEN
    (= :yellow c) CONSOLE_COLOR_YELLOW
    (= :blue c) CONSOLE_COLOR_BLUE
    (= :magenta c) CONSOLE_COLOR_MAGENTA
    (= :cyan c) CONSOLE_COLOR_CYAN
    (= :white c) CONSOLE_COLOR_WHITE
    :else CONSOLE_COLOR_BLACK))

File: typescript/tsfind/src/color.ts
/*
 * color.ts
 *
 * Generic colors
 */

'use strict'

import {ConsoleColor} from "./consolecolor";

export enum Color {
    BLACK = 'black',
    RED = 'red',
    GREEN = 'green',
    YELLOW = 'yellow',
    BLUE = 'blue',
    MAGENTA = 'magenta',
    CYAN = 'cyan',
    WHITE = 'white',
}

export function colorToConsoleColor(color: Color): ConsoleColor {
    switch (color) {
        case Color.BLACK:
            return ConsoleColor.BLACK;
        case Color.RED:
            return ConsoleColor.RED;
        case Color.GREEN:
            return ConsoleColor.GREEN;
        case Color.YELLOW:
            return ConsoleColor.YELLOW;
        case Color.BLUE:
            return ConsoleColor.BLUE;
        case Color.MAGENTA:
            return ConsoleColor.MAGENTA;
        case Color.CYAN:
            return ConsoleColor.CYAN;
        case Color.WHITE:
            return ConsoleColor.WHITE;
    }
}
